chain-impl-mockchain/src/tokens/name.rs

```diff
@@ -60,6 +61,23 @@ impl Readable for TokenName {
     }
 }
 
+impl Bech32 for TokenName {
+    const BECH32_HRP: &'static str = "token-name";
+    const BYTES_LEN: usize = TOKEN_NAME_MAX_SIZE;
```

Comment:
This is not actually the size in bytes of data, but just an upper bound. Since `TokenName` is variable length I think we should use `Bech32m` (see https://github.com/bitcoin/bips/blob/master/bip-0350.mediawiki). There was some discussion already here https://github.com/input-output-hk/chain-libs/pull/630

Reply:
Got it, thanks for this catch.
 But I think an implementation of the `Bech32m` it is not a part of this update and also needs a bigger discussion. I think at this point I will just close this PR, and will use basic hex encoding.